docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/PermutationFeatureImportance.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.ML.Samples.Dynamic.Trainers.Regression
+{
+    public static class PermutationFeatureImportance
+    {
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext(seed:1);
+
+            // Create sample data.
+            var samples = Data.GenerateData();
+
+            // Load the sample data as an IDataView.
+            var data = mlContext.Data.LoadFromEnumerable(samples);
+
+            // Define a training pipeline that concatenates features into a vector, normalizes them, and then
+            // trains a linear model.
+            var pipeline = mlContext.Transforms.Concatenate("Features", Data.FeatureColumns)
+                    .Append(mlContext.Transforms.NormalizeMinMax("Features"))
+                    .Append(mlContext.Regression.Trainers.Ols());
+
+            // Fit the pipeline to the data.
+            var model = pipeline.Fit(data);
+
+            // Compute the permutation metrics for the linear model using the normalized data.
+            var transformedData = model.Transform(data);
+            var linearPredictor = model.LastTransformer;
+            var permutationMetrics = mlContext.Regression.PermutationFeatureImportance(
+                linearPredictor, transformedData, permutationCount: 30);
+
+            // Now let's look at which features are most important to the model overall.
+            // Get the feature indices sorted by their impact on RMSE.
+            var sortedIndices = permutationMetrics.Select((metrics, index) => new { index, metrics.RootMeanSquaredError})
+                .OrderByDescending(feature => Math.Abs(feature.RootMeanSquaredError.Mean))
+                .Select(feature => feature.index);
+
+            Console.WriteLine("Feature\tModel Weight\tChange in RMSE\t95% Confidence in the Mean Change in RMSE");
+            var rmse = permutationMetrics.Select(x => x.RootMeanSquaredError).ToArray();
+            foreach (int i in sortedIndices)
+            {
+                Console.WriteLine("{0}\t{1:0.00}\t{2:G4}\t{3:G4}",
+                    Data.FeatureColumns[i],
+                    linearPredictor.Model.Weights[i],
+                    rmse[i].Mean,
+                    1.96 * rmse[i].StandardError);
+            }
+
+            // Expected output:
+            //  Feature    Model Weight Change in RMSE  95% Confidence in the Mean Change in RMSE
+            //  Feature2        9.00        4.009       0.008304
+            //  Feature1        4.48        1.901       0.003351
```

Comment:
I would align Expected output and Feature.
One space instead of two.
Same across all files.

Reply:
Two spaces for printed verbatim output is the style across the Samples project.

---
In reply to: [273720618](https://github.com/dotnet/machinelearning/pull/3247#discussion_r273720618) [](ancestors = 273720618)